### ISSUE TITLE
Syntax for matching questions

### ISSUE THREAD

**Issue by NiallTracey** [gpoore/text2qti#54]
So the question type I'm most interested in adding is the matching question.

The reason for this is I figure I can make a massive question bank through permutations:
If I have a list of seven definitions I can make questions that give definitions of 3 or 4, and choice of all 7 terms to match with; that in effect gives 70 possible questions (7 choose 3 + 7 choose 4).

I'll never run out of questions!!

Anyhow...

I was thinking maybe something like this

```
1. Match the definition of the left with the word on the right.
A personal transport vehicle typically powered by an internal combustion engine, or increasing an electric motor <=> car
A two-wheeled human-powered vehicle <=> bicycle
! boat
! hovercraft
```

Or do we need an explicit start token for each row?
Not sure about the ! for additional distractors at the end, just what came to mind just now.

**Comment by gpoore** [comment]
I feel like there may need to be some sort of start token for each row. The exclamation point is already being used for solutions/important information, so another character would be needed for distractors. Perhaps something like this would work, since `~` and `x` are unused:
```
1. Match the definition of the left with the word on the right.
~ A personal transport vehicle typically powered by an internal combustion engine, or increasing an electric motor <=> car
~ A two-wheeled human-powered vehicle <=> bicycle
x boat
x hovercraft
```
I like `<=>` as a separator. There may be a potential for collisions, though. I'm hoping to add support for some common shorthand for chemical equations, and `<=>` is used for a double-ended double arrow. One solution would be to use another symbol or some sort of delimiters around the word. Another option would be to add a line break...this avoids ambiguity, but I'm not excited about the extra vertical space.
```
1. Match the definition of the left with the word on the right.
~ A personal transport vehicle typically powered by an internal combustion engine, or increasing an electric motor
=> car
~ A two-wheeled human-powered vehicle
=> bicycle
x> boat
x> hovercraft
```
You could probably go ahead and start implementing something along the lines of what you've suggested, and we can continue discussing syntax.

**Comment by snow-jallen** [comment]
Yes, please!  The ability to write matching questions would be a great improvement.
